diff -aruN nova-2015.1.0.org3/nova/virt/libvirt/config.py nova-2015.1.0/nova/virt/libvirt/config.py
--- nova-2015.1.0.org3/nova/virt/libvirt/config.py	2015-12-02 10:34:42.499539998 +0800
+++ nova-2015.1.0/nova/virt/libvirt/config.py	2015-11-30 09:34:20.795755801 +0800
@@ -1268,6 +1268,8 @@
         self.autoport = True
         self.keymap = None
         self.listen = None
+        #wz for spice password
+        self.passwd = None
 
     def format_dom(self):
         dev = super(LibvirtConfigGuestGraphics, self).format_dom()
@@ -1281,7 +1283,8 @@
             dev.set("keymap", self.keymap)
         if self.listen:
             dev.set("listen", self.listen)
-
+        if self.passwd:
+            dev.set("passwd", self.passwd)
         return dev
 
 
diff -aruN nova-2015.1.0.org3/nova/virt/libvirt/driver.py nova-2015.1.0/nova/virt/libvirt/driver.py
--- nova-2015.1.0.org3/nova/virt/libvirt/driver.py	2015-12-02 10:34:42.499539998 +0800
+++ nova-2015.1.0/nova/virt/libvirt/driver.py	2015-11-30 09:34:12.094888251 +0800
@@ -105,6 +105,8 @@
 from nova.virt import watchdog_actions
 from nova import volume
 from nova.volume import encryptors
+#add by wz cs2c for spice 
+import md5
 
 libvirt = None
 
@@ -4244,6 +4246,9 @@
             graphics.type = "spice"
             graphics.keymap = CONF.spice.keymap
             graphics.listen = CONF.spice.server_listen
+            #wz for spice password
+            graphics.passwd =  md5.new(instance.uuid).hexdigest()
+            LOG.warn('cs2c------------------ %s %s' % (str(graphics),graphics.passwd))
             guest.add_device(graphics)
             add_video_driver = True
 
@@ -4295,7 +4300,7 @@
                 balloon.model = 'xen'
             balloon.period = CONF.libvirt.mem_stats_period_seconds
             guest.add_device(balloon)
-
+        LOG.warn('cs2c------------------ %s' % str(guest))
         return guest
 
     def _get_guest_xml(self, context, instance, network_info, disk_info,
